assets/app/scripts/directives/deleteButton.js

```diff
@@ -0,0 +1,104 @@
+        scope.resourceName = attrs.resourceName;
+        scope.projectName = attrs.projectName;
+        scope.displayName = attrs.displayName;
+
+        if (attrs.resourceType === 'project') {
+          scope.isProject = true;
+          scope.biggerButton = true;
+        }
+
+        // make the resource types show up nicely in the modal
+        scope.nameFormatMap = {
+          'imagestream': 'Image Stream',
+          'pod': 'Pod',
+          'service': 'Service',
+          'buildconfig': 'Build Config',
+          'deploymentconfig': 'Deployment Config',
+          'project': 'Project'
+        };
+
+        scope.routeMap = {
+          'imagestream': 'images',
+          'pod': 'pods',
+          'service': 'services',
+          'buildconfig': 'builds',
+          'deploymentconfig': 'deployments'
+        };
+
+        scope.openDeleteModal = function() {
+          // opening the modal with settings scope as parent
+          var modalInstance = $modal.open({
+            animation: true,
+            templateUrl: 'views/modals/delete-resource.html',
+            controller: 'DeleteModalController',
+            scope: scope
+          });
+
+          modalInstance.result.then(function() {
+          // upon clicking delete button, delete resource and send alert
+            var resourceType = attrs.resourceType;
+            var resourceName = attrs.resourceName;
+            var projectName = attrs.projectName;
+            var formattedResource = scope.nameFormatMap[resourceType] + ' ' + "\'"  + (attrs.displayName ? attrs.displayName : resourceName) + "\'";
+
+            DataService.delete(resourceType + 's', resourceName, scope.$parent)
+            .then(function() {
+              if (resourceType !== 'project') {
+                var browseRedirect = URI('/project/' + projectName + '/browse/' + scope.routeMap[resourceType]);
+                AlertMessageService.addAlert({
+                  name: resourceName,
+                  data: {
+                    type: "success",
+                    message: formattedResource + " was marked for deletion."
+                  }
+                });
+                $location.url(browseRedirect);
+              }
+              else {
+                if ($location.path() === '/') {
+                  scope.$emit('deleteProject');
+                }
+                else if ($location.path().indexOf('settings') > '-1') {
+                  var homeRedirect = URI('/');
+                  AlertMessageService.addAlert({
+                    name: resourceName,
+                    data: {
+                      type: "success",
+                      message: formattedResource + " was marked for deletion."
+                    }
+                  });
+                  $location.url(homeRedirect);
+                }
+              }
+            })
+            .catch(function(err) {
+              // called if failure to delete
+              scope.alerts[resourceName] = {
```

Comment:
this would be the wrong scope for it to show up on the page since this is the scope of the directive not of the controller